extensions/amp-social-share/1.0/social-share.js

```diff
@@ -185,6 +185,20 @@ function getQueryString(endpoint) {
   return endpoint.slice(q, h);
 }
 
+/**
+ * Checks whether or not the userAgent of the current device indicates
+ * that this is an Ios device
+ * @return {boolean}
+ */
+function isIos() {
```

Comment:
Does this need to be a function? Can it be a doc level const instead?

Reply:
IMHO function for now is reasonable since it's only evaluated at the user action time. But I'm still curious how we're planning on testing this.

Reply:
Maybe wrap it in a `once` then? Don't need to evaluate this on every click.

Reply:
So far, I have only done manual testing for this feature.  If that's what you're referring to this is how I tested so far:

I opened the example file: https://github.com/ampproject/amphtml/blob/master/examples/social-share.amp.html

And toggled different devices in the Console (such as iphone).  Then I click on the `sms` or `email` share buttons to verify the behavior and also used a `console.log()` to verify the final `target` value.  This is technically in an AMP environment though.

~~~

I get the idea of evaluating once.  One tradeoff I am considering now is that in the testing method I described above, if we change the device, the `target` actually would not change since it would be defined when the page is loaded.  Maybe it makes sense to evaluate at time of user action then so the `target` would be dynamic and since the function does not appear to be too expensive.  Happy to change it though if people feel strongly about it :).

Reply:
For a user click I really don't feel very strongly that the value should be cached. At this point I look at this as: caching has its costs, especially when we are opening up a wide usage of components.